src/DynamoCore/Linting/Interfaces/IRuleEvaluationResult.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace Dynamo.Linting.Interfaces
+{
+    public interface IRuleEvaluationResult
```

Comment:
do you intend that this interface and related enums is used by anyone but Dynamo? If not, can it be internal?

Reply:
No intention for anyone else but Dynamo to use it.

The issue with making it internal is explained here
https://github.com/DynamoDS/Dynamo/pull/11606#discussion_r618430760

Reply:
@SHKnudsen - is `InitalizeRule` virtual and protected so it can be implemented by third parties or is it just for our use? 

You can make members both internal and protected.

Reply:
`InitializeRule` is just for us.

internal protected seems to have the same issue (https://stackoverflow.com/questions/49250829/c-sharp-protected-internal-method-with-internal-return-type)

However `private protected` seems to do the trick, I was not aware that was a possibility.